eel_hole/utils.py

```diff
@@ -21,11 +46,124 @@ def rst_to_html(rst: str) -> str:
+    - Single line breaks are preserved as literal '\n' inside paragraphs (no <br> tags)
+
+    Args:
+        text (str): Plaintext input string with optional Markdown-style formatting.
+
+    Returns:
+        str: HTML-formatted string.
+    """
+    escaped = html.escape(text)
+
+    # Handle bold and italics (simple Markdown-style)
+    escaped = re.sub(r"\*\*(.+?)\*\*", r"<strong>\1</strong>", escaped)
+    escaped = re.sub(r"\*(.+?)\*", r"<em>\1</em>", escaped)
+
+    paragraphs = re.split(r"\n\s*\n", escaped)
+    html_parts = []
+
+    for para in paragraphs:
+        lines = para.splitlines()
+        html_lines = []
+        in_ul = False
+        in_ol = False
+
+        for line in lines:
+            if re.match(r"^\s*[-*]\s+", line):
+                # Switch from ol to ul if needed
+                if in_ol:
+                    html_lines.append("</ol>")
+                    in_ol = False
+                if not in_ul:
+                    html_lines.append("<ul>")
+                    in_ul = True
+                html_lines.append(f"<li>{line.strip()[2:]}</li>")
+
+            elif re.match(r"^\s*\d+\.\s+", line):
+                # Switch from ul to ol if needed
+                if in_ul:
+                    html_lines.append("</ul>")
+                    in_ul = False
+                if not in_ol:
+                    html_lines.append("<ol>")
+                    in_ol = True
+                html_lines.append(f"<li>{line.strip().split(None, 1)[1]}</li>")
+
+            else:
+                # Close any open lists before adding paragraph text line
+                if in_ul:
+                    html_lines.append("</ul>")
+                    in_ul = False
+                if in_ol:
+                    html_lines.append("</ol>")
+                    in_ol = False
+                html_lines.append(line)  # keep \n inside paragraph as is
+
+        # Close any lists still open at the end of the paragraph
+        if in_ul:
+            html_lines.append("</ul>")
+        if in_ol:
+            html_lines.append("</ol>")
+
+        # Join paragraph lines with newline to preserve single line breaks
+        joined_para = "\n".join(html_lines).strip()
+
+        # Wrap non-list paragraphs in <p>
+        if not (joined_para.startswith("<ul>") or joined_para.startswith("<ol>")):
+            html_parts.append(f"<p>{joined_para}</p>")
+        else:
+            html_parts.append(joined_para)
+
+    return f"<main>\n{'\n'.join(html_parts)}\n</main>"
+
+
+def clean_pudl_resource(resource: Resource, source_key: str) -> Resource:
+    """Clean up the PUDL datapackage descriptions for display.
+
+    PUDL datapackage documentation is all in RST so we use Sphinx machinery to
+    turn it into HTML that our Jinja templates understand.
+    """
+    return Resource(
+        name=resource.name,
+        description=rst_to_html(resource.description),
+        sources=[{"title": source_key}],
```

Comment:
Same same, recommend switching from `sources` to `packages` or the like

Reply:
This one is so named in the [datapackage spec](https://datapackage.org/standard/data-resource/#sources) unfortunately... I do think it's a confusing name though.

We could refactor `__build_search_index` so we don't store the package name on the `Resource` object. But that also touches the display logic since we pass in a list of `Resource`s to the template. So we'd probably make some sort of "resource display" type yada yada. In the end it'll be easier to work with so I'm tempted to do that, but want to make sure this is a rabbit hole we want to go down before I do it.

Reply:
JK this was a shorter rabbit hole than I had feared. Changes incoming.